pwa/app/routes/events.$year.tsx

```diff
@@ -35,19 +36,109 @@ export async function clientLoader({ params }: ClientLoaderFunctionArgs) {
   return await loadData(params);
 }
 
+type EventGroup = {
+  groupName: string;
+  events: Event[];
+};
+
+function groupBySections(events: Event[]): EventGroup[] {
+  // TODO: Handle 2021 remote events
+  const eventsByWeek: Map<number, EventGroup> = new Map();
+  const eventsByChampionship: Map<string, EventGroup> = new Map();
+  const FOCEvents: EventGroup = {
+    groupName: 'FIRST Festival of Champions',
+    events: [],
+  };
+  const preaseasonEvents: EventGroup = { groupName: 'Preseason', events: [] };
+  const offseasonEvents: EventGroup = { groupName: 'Offseason', events: [] };
+  events.forEach((event) => {
+    // Events by week
+    if (event.week != null) {
+      const weekGroup = eventsByWeek.get(event.week);
+      if (weekGroup) {
+        weekGroup.events.push(event);
+      } else {
+        eventsByWeek.set(event.week, {
+          groupName: `Week ${event.week + 1}`,
+          events: [event],
+        });
+      }
+    }
+
+    // Events by Championship
+    if (event.event_type == 3 || event.event_type == 4) {
+      const groupName =
+        event.year >= 2017 && event.year <= 2022
+          ? `FIRST Championship - ${event.city}`
+          : 'FIRST Championship';
+      const championshipGroup = eventsByChampionship.get(groupName);
+      if (championshipGroup) {
+        championshipGroup.events.push(event);
+      } else {
+        eventsByChampionship.set(groupName, {
+          groupName,
+          events: [event],
+        });
+      }
+    }
+
+    // FOC
+    if (event.event_type == 6) {
```

Comment:
This should probably use an `EventType` ported over to TS

Reply:
Oh yeah i hardcoded these for testing but forgot to extract them out

Reply:
on my other repo i just copy pasted the python code into chatgpt and told it what to do